test/functional/test_framework/blocktools.py

```diff
@@ -173,14 +172,18 @@ def create_transaction(node, txid, to_address, *, amount):
 
 def create_raw_transaction(node, txid, to_address, *, amount):
     """ Return raw signed transaction spending the first output of the
-        input txid. Note that the node must be able to sign for the
-        output that is being spent, and the node must not be running
-        multiple wallets.
+        input txid. Note that the node must have a wallet that can sign
+        for the output that is being spent.
     """
-    rawtx = node.createrawtransaction(inputs=[{"txid": txid, "vout": 0}], outputs={to_address: amount})
-    signresult = node.signrawtransactionwithwallet(rawtx)
-    assert_equal(signresult["complete"], True)
-    return signresult['hex']
+    psbt = node.createpsbt(inputs=[{"txid": txid, "vout": 0}], outputs={to_address: amount})
+    for _ in range(2):
+        for w in node.listwallets():
+            wrpc = node.get_wallet_rpc(w)
+            signed_psbt = wrpc.walletprocesspsbt(psbt)
+            psbt = signed_psbt['psbt']
+    final_psbt = node.finalizepsbt(psbt)
+    assert_equal(final_psbt["complete"], True)
+    return final_psbt['hex']
```

Comment:
I haven't been able to wrap my head around this / how this change fits into the greater PR. If you don't mind could you give me a brief explanation or point me in the right direction?

Reply:
This change makes this function use the PSBT workflow rather than the old raw transactions workflow. Additionally, it tries to sign with all of the wallets loaded by the node.

Reply:
> I haven't been able to wrap my head around this / how this change fits into the greater PR. If you don't mind could you give me a brief explanation or point me in the right direction?

My understanding is that legacy wallets support a mix of watch only and spendable scriptpubkeys, while descriptor wallets have to choose between only watch-only or spendable descriptors and refuse to hold both. So while a single wallet worked in many of these tests previously, now two wallets are required to keep all of the same test coverage. In some cases like this the PR creates an extra watchonly wallet to maintain coverage. In other cases it keeps the coverage for legacy wallets but drops it for descriptors to avoid needing the extra wallet.

I think the descriptor watchonly limitation is supposed to be somewhat artificial currently, but in the future if every wallet is required to be fully watchonly or not, then GUI & RPC interfaces may be simplified later.